cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -637,6 +639,65 @@ public Void apply(AppMds mds) throws Exception {
   }
 
   @Override
+  public void addSchedule(final Id.Program program, final ScheduleSpecification scheduleSpecification) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification appSpec = getAppSpecOrFail(mds, program);
+        Map<String, ScheduleSpecification> schedules = Maps.newHashMap(appSpec.getSchedules());
+        schedules.put(scheduleSpecification.getSchedule().getName(), scheduleSpecification);
+        ApplicationSpecification newAppSpec = new AppSpecificationWithChangedSchedules(appSpec, schedules);
+        // TODO: double check this ProgramType.valueOf()
+        replaceAppSpecInProgramJar(program, newAppSpec,
+                                   ProgramType.valueOf(scheduleSpecification.getProgram().getProgramType().name()));
+        mds.apps.updateAppSpec(program.getNamespaceId(), program.getApplicationId(), newAppSpec);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void deleteSchedule(final Id.Program program, final SchedulableProgramType programType,
+                             final String scheduleName) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification appSpec = getAppSpecOrFail(mds, program);
+        Map<String, ScheduleSpecification> schedules = Maps.newHashMap(appSpec.getSchedules());
+        ScheduleSpecification removed = schedules.remove(scheduleName);
+        if (removed == null) {
+          throw new NoSuchElementException("no such schedule @ account id: " + program.getNamespaceId() +
```

Comment:
It will be better to throw a Checked Exception so that caller can take appropriate actions upstream. For instance this is being called from a HttpHandler and if the schedule doesn't exists we should return a 404 and not 500.